### PR TITLE
Update MAINTAINERS to point at the main repo

### PR DESCRIPTION
**Pull request by olix0r**
Update links to refer to the `main` branch.

### CODE REVIEW

**Review by wmorgan — APPROVED** [review]

File: MAINTAINERS.md [hunk 1]
```diff
@@ -1,16 +1 @@
...
-<!--
-# Adding a new maintainer
-
-* Submit a PR modifying this file
-* Add maintainer to .github/CODEOWNERS
-* Obtain approvals per GOVERNANCE.md
-* Invite maintainer to https://github.com/orgs/linkerd/teams/linkerd2-maintainers/members
-* Invite maintainer to https://github.com/orgs/linkerd/people
--->
+See https://github.com/linkerd/linkerd2/blob/main/MAINTAINERS.md
```
Comment (hawkw) [inline MAINTAINERS.md:1]: Take it or leave it: might be worth making this a Markdown link so people can just click it?

File: GOVERNANCE.md [hunk 2]
```diff
@@ -1 +1 @@
-See https://github.com/linkerd/linkerd2/blob/master/GOVERNANCE.md
+See https://github.com/linkerd/linkerd2/blob/main/GOVERNANCE.md
```
Comment (hawkw) [inline GOVERNANCE.md:1]: (similarly, maybe this should also be a link?)

**Review by hawkw — APPROVED** [review]
this looks good to me — couple minor questions about potential changes to the MAINTAINERS.md in the main repo now that it's the source of truth, but no blockers for going ahead and merging this.

File: GOVERNANCE.md (hunk 2 above)
Comment (olix0r) [reply GOVERNANCE.md:1]: hyperlinks are clickable without extra markdown!

**Pull request merged**